flattenObject.js

```diff
@@ -0,0 +1,40 @@
+import isPlainObject from './isPlainObject'
+import forEach from './forEach'
+import filter from './filter'
+/**
+ * This method will flatten the given object onto a single layer,
+ * with keys composed of the keys of each layer separated by a separator.
+ *
+ * @since 5.0.0
+ * @category Object
+ * @param {Object} object The object to flatten.
+ * @param {string} [delimiter='.'] The string to separate the objects' layers.
+ * @returns {Object} Returns  flattened object.
+ * @example
+ *
+ * flattenObject({'a': 1, 'b': {'c': 3, 'd': 4}}, '.')
+ * // => {'a': 1, 'b.c': 3, 'b.d': 4}
+ */
+function flattenObject(object, delimiter) {
```

Comment:
If we don't provide a delimiter here, it defaults to `,` from the `join`. Do we need to set the default value to `.`?  
Otherwise, we might have to update the jsDocs